conda_build/utils.py

```diff
@@ -1780,22 +1780,25 @@ def merge_dicts_of_lists(
     return {k: dol1.get(k, no) + dol2.get(k, no) for k in keys}
 
 
-def prefix_files(prefix):
+def prefix_files(prefix: str | os.PathLike | Path) -> set[str]:
     """
     Returns a set of all files in prefix.
     """
-    res = set()
-    prefix_rep = prefix + os.path.sep
-    for root, dirs, files in walk(prefix):
-        for fn in files:
-            # this is relpath, just hacked to be faster
-            res.add(join(root, fn).replace(prefix_rep, "", 1))
-        for dn in dirs:
-            path = join(root, dn)
-            if islink(path):
-                res.add(path.replace(prefix_rep, "", 1))
-                res.update(expand_globs((path,), prefix))
-    return res
+    prefix = f"{prefix}{os.path.sep}"
+
+    def relpath(*path: str) -> str:
+        # this is os.path.relpath, just hacked to be faster
+        return join(*path).replace(prefix, "", 1).lstrip(os.path.sep)
```

Comment:
Should we move the function out of `prefix_files` to not define on every call?

Reply:
Updated the algorithm slightly and eliminated the need for the helper function entirely, see https://github.com/conda/conda-build/pull/5260/commits/89e2edc26cf03a17daa9dccd381180ba5e4c6d0f